### PR TITLE
feat(SupportWarn): PRD/offtopic warning

### PR DESCRIPTION
**Pull request by DiamondMiner88**
Screenshot_20211121-181057
Screenshot_20211121-181105

### CODE REVIEW

File: Aliucord/src/main/java/com/aliucord/coreplugins/SupportWarn.kt [hunk 1]
```diff
@@ -48,11 +51,20 @@ internal class SupportWarn : Plugin() {
                 chatWrap.visibility = View.GONE
 
                 gateButtonImage.setImageResource(R.e.ic_warning_circle_24dp)
-                gateButtonText.setText("PLEASE READ: This channel is not a support channel, do not ask for help.")
+
+                val text = if (loaded.channelId == PLUGIN_REQUESTS_CHANNEL_ID)
+                    "PLEASE READ: This channel is NOT for requesting plugins, do not send requests here."
+                else "PLEASE READ: This channel is not a support channel, do not ask for help."
```
Comment (TymanWasTaken) [inline Aliucord/src/main/java/com/aliucord/coreplugins/SupportWarn.kt:57]: A when statement would probably be better

**Review by TymanWasTaken — CHANGES_REQUESTED** [review]

**Comment by Vendicated** [comment]
this will keep showing until you accept both dev and prd, no?

File: Aliucord/src/main/java/com/aliucord/coreplugins/SupportWarn.kt [hunk 2]
```diff
@@ -35,36 +45,52 @@ internal class SupportWarn : Plugin() {
         Patcher.addPatch(WidgetChatInput::class.java.getDeclaredMethod("configureChatGuard", ChatInputViewModel.ViewState.Loaded::class.java), Hook {
             val loaded = it.args[0] as ChatInputViewModel.ViewState.Loaded
 
-            if (loaded.channelId in channelList && !loaded.shouldShowVerificationGate && !settings.getBool("devNotSupport", false)) {
-                val binding = bindingMethod(it.thisObject) as WidgetChatInputBinding
+            if (loaded.channelId !in channelList || loaded.shouldShowVerificationGate) return@Hook
+            if (loaded.channelId == PLUGIN_REQUESTS_CHANNEL_ID && settings.getBool("acceptedPrdNotRequests", false)) return@Hook
+            if (settings.getBool("acceptedDevNotSupport", false)) return@Hook
```
Comment (Vendicated) [inline Aliucord/src/main/java/com/aliucord/coreplugins/SupportWarn.kt:51]: fails if channel is prd and user accepted dev channel but not prd

**Review by ItzOnlyAnimal — APPROVED** [review]
deosnt looks like you screwed up your logic this time so LGTM

**Pull request merged**